viewer/packages/data-providers/src/image-360-data-providers/descriptor-providers/events/Cdf360BatchEventCollectionLoader.ts

```diff
@@ -0,0 +1,294 @@
+  Historical360ImageSet<ClassicDataSourceType>[]
+> {
+  private readonly _client: CogniteClient;
+
+  constructor(client: CogniteClient) {
+    super(BATCH_SIZE, BATCH_DELAY_MS);
+    this._client = client;
+  }
+
+  /**
+   * Request descriptors for an events-based collection. This will be batched with other concurrent requests.
+   * Batches are processed sequentially to avoid overwhelming the API.
+   */
+  public async getCollectionDescriptors(
+    metadataFilter: Metadata,
+    preMultipliedRotation: boolean
+  ): Promise<Historical360ImageSet<ClassicDataSourceType>[]> {
+    const siteId = metadataFilter.site_id;
+    if (!siteId) {
+      throw new Error('site_id is required for events-based 360 collections');
+    }
+
+    return this.load({ siteId, preMultipliedRotation });
+  }
+
+  /**
+   * Fetch a batch of event collection descriptors.
+   * This method is called by the base BatchLoader class.
+   */
+  protected async fetchBatch(
+    identifiers: EventCollectionIdentifier[]
+  ): Promise<Map<string, Historical360ImageSet<ClassicDataSourceType>[]>> {
+    // Extract unique site IDs from the batch
+    const siteIds = Array.from(new Set(identifiers.map(id => id.siteId)));
+
+    // Fetch events and files for all site_ids in parallel
+    const [events, files] = await Promise.all([
+      this.listEventsForMultipleSites(siteIds),
+      this.listFilesForMultipleSites(siteIds)
+    ]);
+
+    // Group events and files by site_id
+    const eventsBySiteId = this.groupEventsBySiteId(events);
+    const filesBySiteId = files;
+
+    // Build results map
+    const results = new Map<string, Historical360ImageSet<ClassicDataSourceType>[]>();
+
+    identifiers.forEach(identifier => {
+      const key = this.getKeyForIdentifier(identifier);
+      const siteEvents = eventsBySiteId.get(identifier.siteId) || [];
+      const siteFiles = filesBySiteId.get(identifier.siteId) || new Map();
+
+      const descriptors = this.mergeDescriptors(siteFiles, siteEvents, identifier.preMultipliedRotation);
+      results.set(key, descriptors);
+    });
+
+    return results;
+  }
+
+  /**
+   * Convert an identifier to a string key for result mapping.
+   */
+  protected getKeyForIdentifier(identifier: EventCollectionIdentifier): string {
+    // Include preMultipliedRotation in key since results may differ based on this flag
+    return `${identifier.siteId}:${identifier.preMultipliedRotation}`;
+  }
+
+  /**
+   * Return an empty array when a collection is not found.
+   */
+  protected getDefaultResult(_identifier: EventCollectionIdentifier): Historical360ImageSet<ClassicDataSourceType>[] {
+    return [];
+  }
+
+  private async listEventsForMultipleSites(siteIds: string[]): Promise<CogniteEvent[]> {
+    const allEvents = await Promise.all(
+      siteIds.map(async siteId => {
+        const filter: EventFilter = {
+          metadata: {
+            site_id: siteId
+          }
+        };
+
+        const partitions = 10;
```

Comment:
Should this be a global constant?

Reply:
I'm also wondering if we should use a smaller partition count for the cases where we don't have a lot of 360 images to fetch - but again, it's hard to know up-front 😅 
However - I see that we also have the issue here that we may be sending a very large amount of requests as well